File: c++/leetcode/728. Self Dividing Numbers.cpp
class Solution
{
public:
  vector<int> selfDividingNumbers(int left, int right)
  {
    vector<int> s_divide;

    for (int i = left; i <= right; i++)
    {
      auto digits = i;
      bool self_d = true;

      while (digits)
      {
        int digit = digits % 10;
        if (!digit || i % digit != 0)
        {
          self_d = false;
          break;
        }
        digits /= 10;
      }

      if (self_d)
        s_divide.push_back(i);
    }

    return s_divide;
  }
};

File: javascript/leetcode/728. Self Dividing Numbers.js
/**
 * @param {number} left
 * @param {number} right
 * @return {number[]}
 */
var selfDividingNumbers = function (left, right) {
  const s_divide = [];

  for (let i = left; i <= right; i++) {
    let digits = i;
    let self_d = true;

    while (digits) {
      let digit = digits % 10;
      if (!digit || i % digit !== 0) {
        self_d = false;
        break;
      }
      digits = parseInt(digits / 10);
    }

    if (self_d) s_divide.push(i);
  }

  return s_divide;
};
